fdbserver/BlobManifest.actor.cpp

```diff
@@ -233,6 +216,38 @@ class BlobManifestLoader : public ReferenceCounted<BlobManifestLoader> {
 		return Void();
 	}
 
+	// Iterate active granules and return their version/sizes
+	ACTOR static Future<BlobGranuleVersionVector> listGranules(Reference<BlobManifestLoader> self) {
+		state Transaction tr(self->db_);
+		loop {
+			state BlobGranuleVersionVector results;
+			tr.setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+			tr.setOption(FDBTransactionOptions::PRIORITY_SYSTEM_IMMEDIATE);
+			tr.setOption(FDBTransactionOptions::LOCK_AWARE);
+
+			try {
+				std::vector<KeyRangeRef> granules;
+				state int i = 0;
+				auto limit = GetRangeLimits::BYTE_LIMIT_UNLIMITED;
+				state RangeResult blobRanges = wait(tr.getRange(blobGranuleMappingKeys, limit));
+				for (i = 0; i < blobRanges.size() - 1; i++) {
+					Key startKey = blobRanges[i].key.removePrefix(blobGranuleMappingKeys.begin);
+					Key endKey = blobRanges[i + 1].key.removePrefix(blobGranuleMappingKeys.begin);
+					state KeyRange granuleRange = KeyRangeRef(startKey, endKey);
+					try {
+						Standalone<BlobGranuleVersion> granule = wait(getGranule(&tr, granuleRange));
+						results.push_back_deep(results.arena(), granule);
+					} catch (Error& e) {
+						dprint("missing data for key range {} \n", granuleRange.toString());
```

Comment:
Is this catch intended to swallow the error?

Reply:
It's an error handling case in my back log. In general - if we lose some data files for a granule, should we continue or abort the whole restore? We'll need a separated discussion for this topic.

Reply:
you should only swallow restore_missing_data errors, for regular fdb transaction errors like transaction_too_old you should not print and the transaction should retry

Reply:
fixed. thank you!

for restore_missing_data errors, probably we should report the issue at the end, so that we can know some granules may have inconsistent data...